### ISSUE TITLE
Support syntax highlighting and diff in markdown?

### ISSUE THREAD

**Issue by eddyw** [mattpocock/ts-error-translator#60]
Hey 👋 

Do you think it makes sense to add syntax highlighting support for markdown code?

Also, a more useful feature. Does it maybe make sense to add diff support in markdown? This'd help to easily display inline suggestions of code than should be removed + added. Sort of like GitHub suggestions or diff markdown. For example: 👇 

```diff ts
const foo = "bar";
- type a = (typeof);
+ type a = typeof foo;
```

**Comment by mattpocock** [comment]
Yes, I think we should do syntax highlighting and prettifying for all the code snippets. I.e. you should be able to view `{0}`, `{1}`, etc all pretty, with syntax highlighting.

This is complicated by the fact that TS sometimes has 'and 2 more' in their code snippets. But a smart fella like you could work around that, I'm sure (perhaps wrapping it in a comment, then prettifying it).

I kind of like the idea of diffing (I assume you mean in the body) but not as much as I like the idea of integrating shiki-twoslash for really bringing out the errors. We're using it in the new XState docs, I'm sure it would work great for the site.
